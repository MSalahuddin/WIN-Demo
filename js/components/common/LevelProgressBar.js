import React, { useContext } from 'react';
import styled from 'styled-components/native';
import PropTypes from 'prop-types';
import LinearGradient from 'react-native-linear-gradient';
import { Platform } from 'react-native';
import Text, { SIZE, FONT_FAMILY } from './Text';
import { scale, scaleWidth, scaleHeight, getWindowWidth } from '../../platformUtils';
import { color } from '../../styles';
import { UserContext } from '../../context/User.context';
import { star } from '../../../assets/images';
import { accountProfileStrings } from '../../stringConstants';
import { convertNumberToStringWithComma } from '../../utils';

// container width is the full window minus padding 96 each side
const containerWidth = getWindowWidth() - scaleWidth(162);
const Container = styled.View`
  width: ${containerWidth};
`;

const Row = styled.View`
  align-items: center;
  flex-direction: row;
  justify-content: ${({ isCenter }) => (isCenter ? 'center' : 'space-between')};
  margin-top: ${({ marginTop }) => (marginTop ? scaleHeight(9) : 0)};
`;

const Circle = styled(LinearGradient)`
  align-items: center;
  justify-content: center;
  height: ${scaleHeight(22)};
  width: ${scaleHeight(22)};
  border-radius: ${scaleHeight(18)};
 
`;

const CircleText = styled(Text)`
  color: ${color.black};
  margin-top: ${Platform.OS === 'android' ? scaleHeight(0) : scaleHeight(5)};
  line-height: ${scaleHeight(18)};
 `;

const ProgressBar = styled.View`
  border-radius: ${scale(20)};
  border-width:${scaleHeight(3)};
  background-color: ${color.profileNavigationPrimary};
  height: ${scaleHeight(15)};
  margin-horizontal: ${scale(8)};
  overflow:hidden
  width: 65%;
`;

const Progress = styled.View`
  position: absolute;
  border-radius: ${scale(8)};
  background-color: ${color.profilegradientCircleCurrent2};
  height: ${scaleHeight(9)};
  width: ${({ width }) => width}%;
`;



const PointDisplay = styled(Text)`
  margin-left: ${scaleWidth(5)};
  padding-top: ${Platform.OS === 'android' ? scaleHeight(0) : scaleHeight(6)};
`;

const TextWrapper = styled(Text)`
  margin-top: ${scaleHeight(5)};
`;

const LevelProgressBar = ({ displayProgressInText, ...rest }) => {
  const { vipLevel, vipPoints } = useContext(UserContext);
  const { vipLevelId, nextVipLevel, maxLevelId } = vipLevel;
  const nextLevel = vipLevelId + 1;
  const pointsToNextLevel = nextVipLevel ? nextVipLevel.points : 0;
  const progressWidth = (vipPoints / pointsToNextLevel) * 100;

  const displayPoints = pointsValue => {
    return (
      <PointDisplay color={color.white} size={SIZE.XXSMALL}>
        {pointsValue}
      </PointDisplay>
    );
  };

  const renderText = () => {
    if (displayProgressInText) {
      return (
        <PointDisplay color={color.white} size={SIZE.XXSMALL}>
          {accountProfileStrings.pointsAwayFromLevel(pointsToNextLevel, nextLevel)}
        </PointDisplay>
      );
    }

    if (vipLevelId === maxLevelId) {
      return displayPoints(convertNumberToStringWithComma(vipPoints));
    }

    const pointsComparison = `${convertNumberToStringWithComma(vipPoints)} / ${convertNumberToStringWithComma(
      pointsToNextLevel
    )}`;
    return displayPoints(pointsComparison);
  };

  const renderProgressBar = () => {
    if (vipLevelId === maxLevelId) {
      return (
        <Row isCenter>
          <TextWrapper fontFamily={FONT_FAMILY.SEMI_BOLD} size={SIZE.XSMALL} color={color.silver} alignCenter>
            {accountProfileStrings.maxLevelReached}
          </TextWrapper>
        </Row>
      );
    }

    return (
      <Row isCenter>
        <Circle colors={[color.profilegradientCircleTotal1, color.profilegradientCircleTotal2]}>
          <CircleText size={SIZE.XSMALL} fontFamily={FONT_FAMILY.BOLD}>
            {vipLevelId}
          </CircleText>
        </Circle>
        <ProgressBar>
          <Progress width={progressWidth} />
        </ProgressBar>
        <Circle colors={[color.profilegradientCircleCurrent1, color.profilegradientCircleCurrent2]}>
          <CircleText size={SIZE.XSMALL} fontFamily={FONT_FAMILY.BOLD}>
            {nextLevel}
          </CircleText>
        </Circle>
      </Row>
    );
  };

  return (
    <Container {...rest}>
      <Row isCenter marginTop={displayProgressInText}>
        {renderText()}
        {/* <StarIcon resizeMode="cover" source={star} /> */}
      </Row>
      {renderProgressBar()}
    </Container>
  );
};

LevelProgressBar.propTypes = {
  displayProgressInText: PropTypes.bool
};

LevelProgressBar.defaultProps = {
  displayProgressInText: false
};

export default LevelProgressBar;
