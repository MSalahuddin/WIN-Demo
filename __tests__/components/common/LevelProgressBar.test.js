import React from 'react';
import { render } from 'react-native-testing-library';

import MockProvider from '../../MockProvider';
import LevelProgressBar from '../../../js/components/common/LevelProgressBar';

describe('Component LevelProgressBar', () => {
  test('renders correctly', () => {
    const output = render(
      <MockProvider>
        <LevelProgressBar />
      </MockProvider>
    );
    expect(output).toMatchSnapshot();
  });
});
