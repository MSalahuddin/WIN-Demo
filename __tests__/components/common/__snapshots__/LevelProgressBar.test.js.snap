// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component LevelProgressBar renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "width": 366,
      },
    ]
  }
>
  <View
    isCenter={true}
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 0,
        },
      ]
    }
  >
    <View
      bgColor="#F06B59"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#F06B59",
            "borderRadius": 24,
            "height": 48,
            "justifyContent": "center",
            "width": 48,
          },
        ]
      }
    >
      <Text
        alignCenter={false}
        allowFontScaling={true}
        color="#000000"
        fontFamily="BOLD"
        isUnderlined={false}
        isUppercase={false}
        maxFontSizeMultiplier={1.3}
        size="XSMALL"
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "CalibreBold",
              "fontSize": 24,
              "justifyContent": "center",
              "textAlign": "left",
              "textDecorationColor": "#000000",
              "textDecorationLine": "none",
              "textDecorationStyle": "solid",
              "textTransform": "none",
            },
            Object {
              "color": "#FFFFFF",
              "lineHeight": 38,
              "marginTop": 10,
            },
          ]
        }
      >
        1
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#E5E5E5",
            "borderRadius": 12,
            "height": 16,
            "marginHorizontal": 12,
            "width": "70%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFDB4F",
              "borderRadius": 12,
              "height": 16,
              "position": "absolute",
              "width": "0%",
            },
          ]
        }
        width={0}
      />
    </View>
    <View
      bgColor="#E5E5E5"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#E5E5E5",
            "borderRadius": 24,
            "height": 48,
            "justifyContent": "center",
            "width": 48,
          },
        ]
      }
    >
      <Text
        alignCenter={false}
        allowFontScaling={true}
        color="#000000"
        fontFamily="BOLD"
        isUnderlined={false}
        isUppercase={false}
        maxFontSizeMultiplier={1.3}
        size="XSMALL"
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "CalibreBold",
              "fontSize": 24,
              "justifyContent": "center",
              "textAlign": "left",
              "textDecorationColor": "#000000",
              "textDecorationLine": "none",
              "textDecorationStyle": "solid",
              "textTransform": "none",
            },
            Object {
              "color": "#FFFFFF",
              "lineHeight": 38,
              "marginTop": 10,
            },
          ]
        }
      >
        2
      </Text>
    </View>
  </View>
  <View
    isCenter={true}
    marginTop={false}
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 0,
        },
      ]
    }
  >
    <Image
      resizeMode="cover"
      source={1}
      style={
        Array [
          Object {
            "height": 24,
            "width": 24,
          },
        ]
      }
    />
    <Text
      alignCenter={false}
      allowFontScaling={true}
      color="#353535"
      fontFamily="MEDIUM"
      isUnderlined={false}
      isUppercase={false}
      maxFontSizeMultiplier={1.3}
      size="XXSMALL"
      style={
        Array [
          Object {
            "color": "#353535",
            "fontFamily": "Calibre-Medium",
            "fontSize": 21,
            "justifyContent": "center",
            "textAlign": "left",
            "textDecorationColor": "#353535",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
          Object {
            "marginLeft": 10,
            "paddingTop": 12,
          },
        ]
      }
    >
      0 / 80
    </Text>
  </View>
</View>
`;
